inception-external-search-elastic/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/elastic/ElasticSearchProvider.java

```diff
@@ -61,64 +63,69 @@
     {
         List<ExternalSearchResult> results = new ArrayList<ExternalSearchResult>();
 
-        RestTemplate restTemplate = new RestTemplate();
-
-        ElasticSearchResult queryResult;
-
-        String remoteUrl = aTraits.getRemoteUrl();
         String indexName = aTraits.getIndexName();
-        String searchPath = aTraits.getSearchPath();
+        String hostUrl = aTraits.getRemoteUrl().replaceFirst("https?://", "")
+                .replaceFirst("www.", "")
+                .split(":")[0];
         
-        // Set headers
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
+        Settings settings = Settings.builder()
+                .put("client.transport.ignore_cluster_name", "true")
+                .build();
 
-        // Set query
-        String body;
+        TransportClient client = new PreBuiltTransportClient(settings);
         try {
-            body = prepareQuery(aTraits, aQuery);
+            client.addTransportAddress(
+                    new TransportAddress(
+                            InetAddress.getByName(hostUrl),
+                            9300));
         }
-        catch (JsonProcessingException e) {
-            log.error("Invalid JSON while building search query");
-            return Collections.emptyList();
+        catch (UnknownHostException e) {
+            System.out.println(e.getMessage());
```

Comment:
Please never print and exit ;) Probably best to throw the exception - or slightly worse to do what the previous code did: log and return no results.